packages/authentication/README.md

```diff
@@ -2,45 +2,167 @@
 
 A LoopBack component for authentication support.
 
-# Overview
- It demonstrates how to use LoopBack's user models and passport to interact with other authentication providers.
- User can login using a passport.js strategy, which could include a third party provider.
+**This is a reference implementation showing how to implement an authentication component, it is not production ready.**
 
+## Overview
 
-# Installation
+The component demonstrates how to leverage Passport module and extension points
+provided by LoopBack Next to implement authentication layer.
+
+## Installation
 
 ```shell
 npm install --save @loopback/authentication
 ```
 
-# Basic use
+## Basic use
+
+Start by decorating your controller methods with `@authenticate` to require
+the request to be authenticated.
+
+```ts
+// controllers/my-controller.ts
+import {UserProfile, authenticate} from '@loopback/authentication';
+
+class MyController {
+  constructor(@inject('authentication.user') private user: UserProfile) {}
+
+  @authenticate('BasicStrategy')
+  whoAmI() {
+    return this.user.id;
+  }
+}
+```
 
+Next, implement a Strategy provider to map strategy names specified
+in `@authenticate` decorators into Passport Strategy instances.
+
+```ts
+// providers/auth-strategy.ts
+import {
+  inject,
+  Provider,
+  ValueOrPromise,
+} from '@loopback/context';
+import {
+  BindingKeys,
+  AuthenticationMetadata,
+} from '@loopback/authentication';
+
+import {Strategy} from 'passport';
+import {BasicStrategy} from 'passport-http';
+
+export class MyAuthStrategyProvider implements Provider<Strategy> {
+  constructor(
+    @inject(BindingKeys.Authentication.METADATA)
+    private metadata: AuthenticationMetadata,
+  ) {}
+
+  value() : ValueOrPromise<Strategy> {
+    const name = this.metadata.strategy;
+    if (name === 'BasicStrategy') {
+      return new BasicStrategy(this.verify);
+    } else {
+      return Promise.reject(`The strategy ${name} is not available.`);
+    }
+  }
+
+  verify(username: string, password: string, cb: Function) {
+    // find user by name & password
+    // call cb(null, false) when user not found
+    // call cb(null, userProfile) when user is authenticated
+  }
+}
+```
 
- ```ts
-  const strategy = new BasicStrategy(async (username, password) => {
-  return await findUser(username, password);
-  };
- getAuthenticatedUser(strategy, ParsedRequest);
+In order to perform authentication, we need to implement a custom Sequence
+invoking the authentication at the right time during the request handling.
+
+```ts
+// sequence.ts
+import {
+  FindRoute,
+  inject,
+  InvokeMethod,
+  ParsedRequest,
```

Comment:
Actually, I'll add the missing import as you suggested, so that we can land this pull request sooner.